tests/aliquot/test_aliquot_resources.py

```diff
@@ -0,0 +1,422 @@
+            if k is 'sample_id':
+                continue
+            if k is 'shipment_date':
+                self.assertEqual(parser.parse(aliquot[k]), parser.parse(v))
+            else:
+                self.assertEqual(aliquot[k], v)
+
+        self.assertEqual(2, Aliquot.query.count())
+
+    def test_post_missing_req_params(self):
+        """
+        Test create aliquot that is missing required parameters in body
+        """
+        # Create aliquot data
+        kwargs = {
+            'external_id': 'AL1',
+            'shipment_origin': 'CORIELL',
+            'shipment_destination': 'Baylor',
+            'analyte_type': 'DNA',
+            'concentration': 200,
+            'volume': 13.99
+        }
+        # Send post request
+        response = self.client.post(url_for(ALIQUOTS_LIST_URL),
+                                    headers=self._api_headers(),
+                                    data=json.dumps(kwargs))
+
+        # Check status code
+        self.assertEqual(response.status_code, 400)
+
+        # Check response body
+        response = json.loads(response.data.decode("utf-8"))
+
+        # Check error message
+        message = 'could not create aliquot'
+        self.assertIn(message, response['_status']['message'])
+
+        # Check db
+        al = Aliquot.query.first()
+        self.assertIs(al, None)
+
+    # def test_custom_datetime_field(self):
+    #     """
+    #     Test aliquot custom fields datetime
+    #     """
+    #     kwargs = self._create_save_to_db()
+    #
+    #     datetimes = {
+    #         'valid': ['12/20/23', '12-20-2023', 'December 20, 2023'],
+    #         'invalid': [10, 0, 'hai der']
+    #     }
+    #
+    #     # Create aliquot data
+    #     kwargs = {
+    #         'external_id': 'AL1',
+    #         'shipment_origin': 'CORIELL',
+    #         'shipment_destination': 'Baylor',
+    #         'analyte_type': 'DNA',
+    #         'concentration': -200,
+    #         'volume': -13.99,
+    #         'sample_id': kwargs.get('sample_id')
+    #     }
+    #
+    #     for dt in datetimes['valid']:
+    #         # Send post request
+    #         response = self.client.post(url_for(ALIQUOTS_LIST_URL),
+    #                                     headers=self._api_headers(),
+    #                                     data=json.dumps(kwargs))
+    #
+    #     # Check status code
+    #     self.assertEqual(response.status_code, 400)
+    #
+    #     # Check response body
+    #     response = json.loads(response.data.decode("utf-8"))
+    #
+    #     # Check error message
+    #     message = 'could not create aliquot'
+    #     self.assertIn(message, response['_status']['message'])
+    #     errors = ['positive float', 'positive integer']
+    #     for e in errors:
+    #         self.assertIn(e, response['_status']['message'])
+    #
+    #     # Check db
+    #     self.assertIs(1, Aliquot.query.count())
+
+    def test_post_negative_quantities(self):
```

Comment:
Could also be part of an invalid input test in test_api

Reply:
👍